### PR TITLE
[5.2] Add image_data_uri() helper.

### PR DESCRIPTION
**Pull request by yappkahowe**
Add image_data_uri() helper to convert an image to base64 encoded data uri.

### CODE REVIEW

File: src/Illuminate/Support/helpers.php [hunk 1]
```diff
@@ -559,6 +559,27 @@ function head($array)
     }
 }
 
+if (! function_exists('image_data_uri')) {
+    /**
+     * Convert an image to data URI.
+     * 
+     * @param  string $image
+     * @return string
```
Comment (crynobone) [inline src/Illuminate/Support/helpers.php]: it could actually return `null` if `$image` is not an image.

Comment (BrandonSurowiec) [inline src/Illuminate/Support/helpers.php]: You can do your `return` directly in the `if()` statement and get rid of the `$data_uri` variable. No return statement is also `NULL`.

File: src/Illuminate/Support/helpers.php [hunk 2]
```diff
@@ -559,6 +559,27 @@ function head($array)
...
+     */
+    function image_data_uri($image)
+    {
+        $data_uri = null;
+
+        if (file_exists($image) && ($image_type = exif_imagetype($image))) {
+            $mime_type = image_type_to_mime_type($image_type);
+            $data = base64_encode(file_get_contents($image));
+            $data_uri = "data:$mime_type;base64,$data";
```
Comment (BrandonSurowiec) [inline src/Illuminate/Support/helpers.php]: You can do your `return` directly in the `if()` statement and get rid of the `$data_uri` variable. No return statement is also `NULL`.

**Comment by barryvdh** [comment]
This seems a bit specific to be a global helper..

**Comment by taylorotwell** [comment]
I don't think I'm going to include this in the core at this time. Would rather let developers define it in their own helper functions.

**Pull request closed without merging**